src/gflags.cc

```diff
@@ -800,7 +826,7 @@ CommandLineFlag* FlagRegistry::SplitArgumentLocked(const char* arg,
                                     kError, key->c_str());
       return NULL;
     }
-    if (strcmp(flag->type_name(), "bool") != 0) {
+    if (!flag->OfType<bool>()) {
```

Comment:
That will require us to make class FlagValueTraits or enum FlagValue::ValueType to be public.
If that is OK, I'll make that, but at the moment both of them are private.